### PR TITLE
Attach the route summarization policy under L3Out Ext EPG

### PR DESCRIPTION
**Pull request by devegupt**
Adding support to attach the route summarization policy under L3Out External EPG Subnet

### CODE REVIEW

**Comment by andbyrne** [comment]
This appears to be a breaking change. The original schema should be retained, and continue to apply the default policy from the common tenant if no policy is specified in the new `bgp_route_summarization_policy` attribute.

**Comment by devegupt** [comment]
> This appears to be a breaking change. The original schema should be retained, and continue to apply the default policy from the common tenant if no policy is specified in the new `bgp_route_summarization_policy` attribute.

Sounds good, I am working on this.

**Comment by devegupt** [comment]
> This appears to be a breaking change. The original schema should be retained, and continue to apply the default policy from the common tenant if no policy is specified in the new `bgp_route_summarization_policy` attribute.

I have pushed the updates in this PR. As per the new logic:

1. If "bgp_route_summarization" is set to 'true' and "bgp_route_summarization_policy" is not defined, 'default' route summarization policy will be added.
2. If "bgp_route_summarization" is set to 'true' and "bgp_route_summarization_policy" is defined, custom policy name will be added.
3. If "bgp_route_summarization" is set to 'false' and "bgp_route_summarization_policy" is defined, no policy will be added.
4. If "bgp_route_summarization" is set to 'false' and "bgp_route_summarization_policy" is not defined, no policy will be added.

File: modules/terraform-aci-external-endpoint-group/variables.tf [hunk 1]
```diff
@@ -115,6 +115,7 @@ variable "subnets" {
     aggregate_export_route_control = optional(bool, false)
     aggregate_shared_route_control = optional(bool, false)
     bgp_route_summarization        = optional(bool, false)
+    bgp_route_summarization_policy = string
```
Comment (juchowan) [inline modules/terraform-aci-external-endpoint-group/variables.tf:118]: should be optional

**Review by juchowan — CHANGES_REQUESTED** [review]

File: modules/terraform-aci-external-endpoint-group/variables.tf (hunk 1 above)
Comment (devegupt) [reply modules/terraform-aci-external-endpoint-group/variables.tf]: Good catch, fixed.

**Comment by devegupt** [comment]
> This appears to be a breaking change. The original schema should be retained, and continue to apply the default policy from the common tenant if no policy is specified in the new `bgp_route_summarization_policy` attribute.

done

File: modules/terraform-aci-external-endpoint-group/variables.tf [hunk 2]
```diff
@@ -115,6 +115,7 @@ variable "subnets" {
     aggregate_export_route_control = optional(bool, false)
     aggregate_shared_route_control = optional(bool, false)
     bgp_route_summarization        = optional(bool, false)
+    bgp_route_summarization_policy = optional(string)
```
Comment (juchowan) [inline modules/terraform-aci-external-endpoint-group/variables.tf:118]: with the default "" value

**Review by juchowan — CHANGES_REQUESTED** [review]

**Comment by juchowan** [comment]
please also run pre-commit

File: modules/terraform-aci-external-endpoint-group/variables.tf (hunk 2 above)
Comment (devegupt) [reply modules/terraform-aci-external-endpoint-group/variables.tf]: fixed

**Comment by devegupt** [comment]
> please also run pre-commit

Done

**Pull request merged**